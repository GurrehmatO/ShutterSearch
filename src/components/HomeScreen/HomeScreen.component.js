import React, { Component } from 'react';
import {
  View, FlatList, Keyboard, TouchableHighlight, ActivityIndicator,
} from 'react-native';
import {
  Text, SearchBar, Button, Image,
} from 'react-native-elements';
import {
  APP_NAME, HOME_HEADING, SEARCH_PLACEHOLDER, SEARCH_BUTTON,
} from '../../constants/strings';
import getImages from '../../utils/getImages.util';
import styles from './HomeScreen.component.style';

class HomeScreen extends Component {
  static navigationOptions = {
    title: APP_NAME,
  };

  state = {
    searchKey: '',
    pages: 0,
    images: [],
  }

  onChangeText = (value) => {
    this.setState({
      searchKey: value,
      pages: 0,
      images: [],
    });
  }

  onSearch = () => {
    const { searchKey } = this.state;
    Keyboard.dismiss();
    getImages(1, searchKey)
      .then(({ data }) => {
        this.setState({
          pages: 1,
          images: data.data,
        });
      })
      .catch(console.error);
  }

  renderImage = ({ item }) => {
    const { navigation } = this.props;
    return (
      <TouchableHighlight
        style={styles.imageContainer}
        onPress={() => {
          navigation.navigate('FullImage', { item });
        }}
      >
        <Image
          source={{ uri: item.assets.huge_thumb.url }}
          PlaceholderContent={(<ActivityIndicator />)}
          style={styles.image}
        />
      </TouchableHighlight>
    );
  }

  getKey = item => item.id;

  onReachingEnd = () => {
    const { searchKey, pages, images } = this.state;
    getImages(pages + 1, searchKey)
      .then(({ data }) => {
        this.setState({
          pages: pages + 1,
          images: [
            ...images,
            ...data.data,
          ],
        });
      });
  }

  render() {
    const { searchKey, images, pages } = this.state;
    return (
      <View
        style={styles.HomeContainer}
        onPress={() => Keyboard.dismiss()}
      >
        <Text h2 style={styles.Heading}>{HOME_HEADING}</Text>
        <View style={styles.SearchContainer}>
          <SearchBar
            style={styles.SearchBar}
            lightTheme
            placeholder={SEARCH_PLACEHOLDER}
            containerStyle={styles.SearchBarContainer}
            inputStyle={styles.SearchText}
            value={searchKey}
            onChangeText={this.onChangeText}
          />
          <Button
            title={SEARCH_BUTTON}
            buttonStyle={styles.button}
            disabled={(searchKey === '')}
            onPress={this.onSearch}
          />
        </View>
        <View style={styles.resultContainer}>
          {(pages > 0) && (
          <FlatList
            data={images}
            horizontal={false}
            numColumns={3}
            renderItem={this.renderImage}
            keyExtractor={this.getKey}
            onEndReachedThreshold={0.5}
            onEndReached={this.onReachingEnd}
          />
          )}
        </View>
      </View>
    );
  }
}

export default HomeScreen;
