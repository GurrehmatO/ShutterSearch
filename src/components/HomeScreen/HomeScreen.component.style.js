import { Dimensions } from 'react-native';

const styles = {
  HomeContainer: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'flex-start',
  },
  Heading: {
    width: '100%',
    textAlign: 'center',
  },
  SearchContainer: {
    width: '100%',
  },
  SearchBar: {
    width: '100%',
    borderColor: 'transparent',
    marginBottom: 20,
  },
  SearchBarContainer: {
    backgroundColor: 'transparent',
    borderColor: 'transparent',
  },
  SearchText: {
    color: 'white',
  },
  button: {
    width: '50%',
    alignSelf: 'center',
    marginTop: 16,
    marginBottom: 20,
  },
  resultContainer: {
    flex: 1,
    width: '100%',
  },
  imageContainer: {
    margin: 2,
    flex: 1,
  },
  image: {
    width: (Dimensions.get('window').width / 3) - 5,
    height: (Dimensions.get('window').width / 3) - 5,
  },
};

export default styles;
