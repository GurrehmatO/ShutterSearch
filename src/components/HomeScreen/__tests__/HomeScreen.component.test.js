import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../HomeScreen.component';

describe('HomeScreen component', () => {
  it('should render correctly', () => {
    const component = renderer.create(<HomeScreen />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
