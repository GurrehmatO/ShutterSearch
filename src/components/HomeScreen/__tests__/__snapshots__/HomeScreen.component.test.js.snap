// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen component should render correctly 1`] = `
<View
  onPress={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 42.5,
        "textAlign": "center",
        "width": "100%",
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  >
    Lets search ShutterStock
  </Text>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "#e1e1e1",
          "borderBottomWidth": 1,
          "borderColor": "transparent",
          "borderTopColor": "#e1e1e1",
          "borderTopWidth": 1,
          "padding": 8,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 0,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdc6cf",
              "borderBottomWidth": 0,
              "borderColor": "#86939e",
              "borderRadius": 3,
              "flexDirection": "row",
              "minHeight": 30,
              "overflow": "hidden",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 8,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#86939e",
                        "fontSize": 18,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  î¢¶
                </Text>
              </View>
            </View>
          </View>
          <TextInput
            allowFontScaling={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onClear={[Function]}
            onFocus={[Function]}
            placeholder="Enter your search term"
            placeholderTextColor="#86939e"
            platform="default"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "white",
                "flex": 1,
                "fontSize": 18,
                "marginLeft": 10,
                "minHeight": 40,
              }
            }
            testID="searchInput"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 15,
                "marginRight": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "hsl(208, 8%, 90%)",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 20,
              "marginTop": 16,
              "padding": 8,
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": Object {
                  "color": Array [
                    208,
                    8,
                    63,
                  ],
                  "model": "hsl",
                  "valpha": 1,
                },
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Search
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "width": "100%",
      }
    }
  />
</View>
`;
